packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -61,17 +61,17 @@ export function createInsightsMiddleware<
     $$internal = false,
   } = props;
 
-  let insightsClient: InsightsClientWithGlobals = _insightsClient || noop;
+  let potentialInsightsClient: ProvidedInsightsClient = _insightsClient;
 
-  if (_insightsClient !== null && !_insightsClient) {
+  if (!_insightsClient && _insightsClient !== null) {
     safelyRunOnBrowser(({ window }: { window: any }) => {
       const pointer = window.AlgoliaAnalyticsObject || 'aa';
 
       if (typeof pointer === 'string') {
-        insightsClient = window[pointer];
+        potentialInsightsClient = window[pointer];
       }
 
-      if (!insightsClient) {
+      if (!potentialInsightsClient) {
```

Comment:
before this change, if window.AlgoliaAnalyticsObject wasn't a string, `insightsClient` still was `noop`, and thus not null